components/public-api/typescript-common/src/public-api-converter.ts

```diff
@@ -236,6 +237,56 @@ export class PublicAPIConverter {
         return result;
     }
 
+    toWorkspaceSessionMetrics(status: WorkspaceInstanceStatus, metrics: WorkspaceInstanceMetrics | undefined): WorkspaceSession_Metrics {
```

Comment:
Amazing that you found a clean way to account for older workspaces with metrics on their status. Kudos!